src/Mantasflowers.WebApi/Controllers/OrdersController.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Mantasflowers.Contracts.Errors;
+using Mantasflowers.Contracts.Order.Response;
+using Mantasflowers.Services.Services.Order;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+
+namespace Mantasflowers.WebApi.Controllers
+{
+    [ApiController]
+    [Route("/order")]
+    [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ErrorResponse), StatusCodes.Status500InternalServerError)]
+    public class OrdersController : ControllerBase
+    {
+        private readonly IOrderService _orderService;
+
+        public OrdersController(IOrderService orderService)
+        {
+            _orderService = orderService;
+        }
+
+        [Authorize]
+        [HttpGet("{id}")]
+        [ProducesResponseType(typeof(GetDetailedOrderResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> GetOrder(Guid id)
```

Comment:
We aren't checking if order belongs to the user. On a separate note: what is the situation with `UserOrders` ? Don't really understand how orders are being connected to users now.

Reply:
At the moment, they aren't. To associate with a user, we'd have to receive a token and have a separate logic/endpoint(?) for this use case.

Reply:
Might be the case (like /attach-order) or smth. Since orders might not actually have a registered user.